src/main/java/com/google/sps/data/TimeRange.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main.java.com.google.sps.data;
+
+import java.time.Instant;
+import java.time.Duration;
+import java.util.Comparator;
+
+/**
+ * Class representing a span of time, enforcing properties (e.g. start comes before end) and
+ * providing methods to make ranges easier to work with (e.g. {@code overlaps}).
+ */
+public final class TimeRange {
+
+  /** A comparator for sorting ranges by their start time in ascending order. */
+  public static final Comparator<TimeRange> ORDER_BY_START =
+      new Comparator<TimeRange>() {
+        @Override
+        public int compare(TimeRange a, TimeRange b) {
+          return a.start.compareTo(b.start);
+        }
+      };
+
+  /** A comparator for sorting ranges by their end time in ascending order. */
+  public static final Comparator<TimeRange> ORDER_BY_END =
+      new Comparator<TimeRange>() {
+        @Override
+        public int compare(TimeRange a, TimeRange b) {
+          return a.end().compareTo(b.end());
+        }
+      };
+
+  private final Instant start, end;
+
+  /** Creates a timerange with a start and end instant */
+  private TimeRange(Instant start, Instant end) {
+    this.start = start;
+    this.end = end;
+  }
+
+  /** Returns the start of the range as an instant. */
+  public Instant start() {
+    return this.start;
+  }
+
+  /** Returns the number of minutes between the start and end of the range. */
+  public Long duration() {
```

Comment:
Might want to leave it as an int so we don't have to keep casting it to long's later on. @acmcarther ?

Reply:
I'd suggest returning the Duration. If caller wants minutes, they can get minutes where they need it.

Passing around times or durations as primitive types is a recipe for unit conversion errors.